File: JamTime.Backend/src/JamTime.Backend.Web/Endpoints/Authentication/TokenData.cs
namespace JamTime.Backend.Web.Endpoints.Authentication;

public class TokenData
{
  public string AccessToken { get; set; } = default!;
  public DateTime AccessTokenExpirationTime { get; set; }
  public string RefreshToken { get; set; } = default!;
  public DateTime RefreshTokenExpirationTime { get; set; }
}

File: jam-time-frontend/src/app/core/services/auth/token-data.ts
export class TokenData {
  public accessToken: string;
  public accessTokenExpirationTime: Date;
  public refreshToken: string;
  public refreshTokenExpirationTime: Date;

  public constructor(accessToken: string, accessTokenExpirationTime: Date, refreshToken: string, refreshTokenExpirationTime: Date) {
    this.accessToken = accessToken;
    this.accessTokenExpirationTime = accessTokenExpirationTime;
    this.refreshToken = refreshToken;
    this.refreshTokenExpirationTime = refreshTokenExpirationTime;
  }
}
